jujugui/static/gui/src/app/store/env/go.js

```diff
@@ -2442,6 +2443,155 @@ YUI.add('juju-env-go', function(Y) {
     },
 
     /**
+      Make service endpoints available for consumption.
+
+      @method offer
+      @param {String} service The name of the service being offered.
+      @param {Array of strings} endpoints The offered endpoint names.
+      @param {String} url The URL to use to reference the resulting remote
+        service. For instance "local:/u/admin/ec2/django".
+        If empty, a URL is automatically generated using the
+        "local:/u/$user/$env-name/$service" pattern.
+      @param {Array of strings} users Users that these endpoints are offered
+        to. If left empty, the offer is considered public.
+      @param {String} description Description for the offered service. It
+        defaults to the description provided in the charm.
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive an object with the following
+        attributes:
+        - err: optional error, only defined if something went wrong;
+        - service: the provided name of the offered service;
+        - endpoints: the provided offered endpoints;
+        - url: the offered service URL.
+    */
+    offer: function(service, endpoints, url, users, description, callback) {
+      // Generate the URL if an empty one has been provided.
+      // XXX frankban 2015/12/15: this will be done automatically by the
+      // server, and the URL will be returned as part of the API response.
+      if (!url) {
+        var user = this.getCredentials().user.replace(/^user-/, '');
+        var envName = this.get('environmentName');
+        url = 'local:/u/' + user + '/' + envName + '/' + service;
+      }
+
+      // Define the API callback.
+      var handleOffer = function(userCallback, service, endpoints, url, data) {
+        if (!userCallback) {
```

Comment:
We need to handle the case a callback is not passed, and I think it's good to have at least a log of what we are discarding.